sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -262,12 +264,36 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
     }.flatten
 
     // List of SQL queries to run
-    // note: this is not a robust way to split queries using semicolon, but works for now.
-    val queries = (importedCode ++ code).mkString("\n").split("(?<=[^\\\\]);")
```

Comment:
I think we should use stronger delimiter that is not ambiguous. e.g. a line of `--QUERY-DELIMITER`. When we detect that a test file contains `--QUERY-DELIMITER`, we use it to split queries instead of semicolon.

Reply:
OK. A better method. I will try on this way!

Reply:
Plz update how-to-use it in the top? https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L66-L77

Reply:
Based on the @cloud-fan suggestion, I thought it like this;
```
--QUERY-DELIMITER ':'
some test query:
some test query:
...
--QUERY-DELIMITER ';'
```
I think It is hard to see what's a delimiter now from the name `--QUERY-DELIMITER-START`. Or, any concern?

Reply:
Because use `--QUERY-DELIMITER-START` and `--QUERY-DELIMITER-END` can make code easier. Because SQL and comments could contains ':' or ';', so use ':' or ';' can't works well.

Reply:
I will update how to use it in the top of `SQLQueryTestSuite`.